crates/threshold-bls/src/sig/sig.rs

```diff
@@ -65,28 +65,44 @@ pub trait SignatureScheme: Scheme {
     fn verify(public: &Self::Public, msg: &[u8], sig: &[u8]) -> Result<(), Self::Error>;
 }
 
-/// Extension trait over `SignatureScheme` which provides signing & verification methods
-/// which do not hash the message.
-pub trait SignatureSchemeExt: SignatureScheme {
-    /// Signs the message with the provided private key and returns a serialized signature. This
-    /// method **will not** hash the message before signing it. It should be used for
-    /// blind-signature related functionalities. In other cases, prefer `Signature`
-    fn sign_without_hashing(private: &Self::Private, msg: &[u8]) -> Result<Vec<u8>, Self::Error>;
-
-    /// Verifies that the signature on the provided message was produced by the public key.
-    /// This method **will not** h the message. It should be used when verifying blind signatures
-    /// by parties that do not have access to the blinding factor
-    fn verify_without_hashing(
-        public: &Self::Public,
-        msg: &[u8],
-        sig: &[u8],
-    ) -> Result<(), Self::Error>;
-}
-
-/// Blinder holds the functionality of blinding and unblinding a message. It is
-/// not to be used alone but in combination with a signature scheme or a
-/// threshold scheme.
-pub trait Blinder {
+/// BlindScheme is a signature scheme where the message can be blinded before
+/// signature so the signer does not know the real message. The signature can
+/// later be "unblinded" as to reveal a valid signature over the initial
+/// message.
+///
+/// ```
+///  # #[cfg(feature = "bls12_381")]
+///  # {
+///  use rand::prelude::*;
+///  use threshold_bls::{
+///     sig::{BlindScheme,SignatureScheme, Scheme, G2Scheme},
+///     group::{Element, Point}
+///  };
+///  use threshold_bls::curve::bls12381::PairingCurve as PC;
+///
+///  let msg = vec![1,9,6,9];
+///  let (private,public) = G2Scheme::<PC>::keypair(&mut thread_rng());
```

Comment:
If you want, you can `use threshold_bls::schemes::bls12_381::G1Scheme as SigScheme` instead. The `schemes` module contains preinstantiated G1/G2 schemes for each curve. Also, could you add spaces after each comma `,` in the function calls?